Optimize performance by using isset() instead of in_array()

andrey-legayev opened the pull request:
### Description (*)
I've run XDebug profiler to see why static content deploy takes so much time. 
Many calls of in_array() grabbed my attention and I've fixed two such places:
1. lib/internal/Magento/Framework/Module/ModuleList/Loader.php 
2. lib/internal/Magento/Framework/Setup/Lists.php 

### Fixed Issues (if relevant)
N/A

### Manual testing scenarios (*)
General regression testing.

### Questions or comments
N/A

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

andrey-legayev commented:
I need a help with following code issue:
```
FILE: ...html/lib/internal/Magento/Framework/Module/ModuleList/Loader.php
----------------------------------------------------------------------
FOUND 0 ERRORS AND 2 WARNINGS AFFECTING 2 LINES
----------------------------------------------------------------------
 203 | WARNING | Direct throw of generic Exception is discouraged.
     |         | Use context specific instead.
```

@ihor-sviziev do you have an idea which exception should be thrown instead of \Exception ?

ihor-sviziev commented on lib/internal/Magento/Framework/Module/ModuleList/Loader.php:
```diff
@@ -196,18 +196,19 @@ private function prearrangeModules(array $modules): array
         }
-        return $result;
+        $allResults[] = $result;
+        return array_unique(call_user_func_array('array_merge', $allResults));
```
```suggestion
        return array_unique(array_merge(...$allResults));
```

ihor-sviziev reviewed (changes requested):
Hi @andrey-legayev,
I think \Exception there could be replaced to \LogicException

magento-engcom-team commented:
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-5923](https://jira.corp.magento.com/browse/ENGCOM-5923) has been created to process this Pull Request
:eight_spoked_asterisk: @ihor-sviziev, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

andrey-legayev committed:
Optimize modules filtering in Magento\Framework\Module\ModuleList\Loader::load()

andrey-legayev committed:
Optimize performance by using isset() instead of in_array()

I've run XDebug profiler to see why static content deploy takes so much
time. Many in_array() calls grabbed my attention and I've fixed it.

Co-Authored-By: Ihor Sviziev <ihor-sviziev@users.noreply.github.com>

andrey-legayev commented:
Updates:
- fixed merge conflict
- replaced one more in_array inside loop by isset()

magento-engcom-team commented:
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-5923](https://jira.corp.magento.com/browse/ENGCOM-5923) has been created to process this Pull Request
